### DISCUSSION TITLE
Flux custom icon(s)

### DISCUSSION THREAD

**Discussion by lcallif** [livewire/flux#610] [Feature Requests]
The ability to create custom icons is great.  However, it creates some issues.  Keeping track of all of the icons we create via flux:icon, ensuring that we don't have any of our custom icons overwritten by Flux if they happen to be the same name, or any other scenario I may have missed.  We might be able to change the flux components (accordion, button, etc, to use the directory containing the icons we want to be used, but that is an issue when y'all update or create new flux components.

The Icons that we create, we can change the name with something like a prefix and keep it in the flux/icon directory.  Somewhat schlocky, but it works.

There has to be a better way to add new icons within Flux that helps ensure that we don't step on each others toes. Maybe something like a Flux icon "custom" directory (flux/icon/custom) where all the icons created by us via the flux:icon would be created at and may be specified via a variant passed to the component.....

**Comment by jeffchown** [comment]
@lcallif I agree that it is important to have a way to keep things organized and separate from 'standard' Flux components and icons to avoid collisions and unexpected overwrites.

After reading your post, I tried creating my own custom icon, using the template provided here: https://fluxui.dev/components/icon#custom-icons

I created my custom icon file in the `resources/views/flux/icon/custom` directory (per your comment) and I was then able to use my new icon via `<flux:icon.custom.doctor />`.  

So it looks like we do have the ability to organize our custom, manually created, icons to avoid collision and overwrites during Flux updates, etc.

This does not address any icons created/imported from Lucide via e.g. `php artisan flux:icon crown grip-vertical github`, but we can manually move the created files to the `/custom` subdirectory for now.

**Reply by lcallif** [reply]
This resolves my issue well enough for me to consider this closed.  Thank you!!

**Reply by jeffchown** [reply]
You're welcome, @lcallif !
